.circleci/config.yml

```diff
@@ -291,6 +298,30 @@ commands:
           name: Wait for local UI server to start
           command: dockerize -wait tcp://localhost:4200 -timeout 4m
 
+  e2e_skip_commit:
+    description: "Halt job if commit message contains 'SKIP E2E'"
+    steps:
+      - run:
+          name: "Halt job if commit message contains 'SKIP E2E'"
+          command: |
+            echo 'export COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")' >> $BASH_ENV
+            source $BASH_ENV
+            echo "Git commit message: $COMMIT_MESSAGE"
+            # Double comma is called "Parameter Expansion" in Bash version 4+. It converts string to lowercase letters.
+            if [[ "${COMMIT_MESSAGE,,}" == *"skip e2e"* ]]; then
+              echo "Halting e2e test."
+              circleci step halt
+            fi
+
+
+  e2e_presubmit_check:
+    steps:
+      - e2e_skip_commit
+      - halt_if_code_unchanged
+      - run:
+          name: "Halt job if not a pull request"
+          command: bash .circleci/pr-skip-ci.sh
```

Comment:
Do  all the halt commnads run here?

Also, can we  mark the test as not run instead of passed or failed somehow? (scope creep, I know). There's a nice gray X or something  in the UI, but I don't  remember what triggered it.